XamlControlsGallery/Navigation/NavigationRootPage.xaml.cs

```diff
@@ -36,17 +36,26 @@ namespace AppUIBasics
 {
     public sealed partial class NavigationRootPage : Page
     {
-        public static NavigationRootPage Current;
-        public static Frame RootFrame = null;
-
         public Windows.System.VirtualKey ArrowKey;
+        public Frame RootFrame { get { return rootFrame; }}
 
         private RootFrameNavigationHelper _navHelper;
         private bool _isGamePadConnected;
         private bool _isKeyboardConnected;
         private Microsoft.UI.Xaml.Controls.NavigationViewItem _allControlsMenuItem;
         private Microsoft.UI.Xaml.Controls.NavigationViewItem _newControlsMenuItem;
 
+        public static NavigationRootPage GetForElement(object obj)

```

Comment:
>object [](http://example.com/codeflow?start=55&length=6)

Should this be typed as UIElement?

Reply:
No, since it supports Application objects as well.  (App.xaml.cs:253)

Reply:
But this will fail immediately if passed anything other than a UIElement, since the very first line does a cast.